shell/platform/embedder/embedder.h

```diff
@@ -2601,9 +2615,18 @@ FlutterEngineResult FlutterEngineAddView(FLUTTER_API_SYMBOL(FlutterEngine)
 //------------------------------------------------------------------------------
 /// @brief      Removes a view.
 ///
-///             This is an asynchronous operation. The view's resources must not
-///             be cleaned up until the |remove_view_callback| is invoked with
-///             a |removed| value of `true`.
+///             This operation notifies the engine to deallocate resources and
+///             inform Dart about the removal. Finally, it invokes
+///             |info.remove_view_callback| with whether the operation is
+///             successful.
+///
+///             This operation is asynchronous. The embedder should not
+///             deallocate resources until the |callback| is invoked.
```

Comment:
The previous wording was more precise as to when the embedder can destroy a view's resources:

> The view's resources must not be cleaned up until the |remove_view_callback| is invoked with a |removed| value of `true`

This is important as the engine can continue presenting to the view if `FlutterEngineRemoveView` returned something other than `kSuccess`. Consider:

```suggestion
///             deallocate resources until the |info.remove_view_callback| is invoked with a |removed| value of `true`.
```